src/cacheRefresher.js

```diff
@@ -0,0 +1,200 @@
+        */
+        constructor(cache, client, cacheSchema, rootKey) {
+            const connectionParameters = client._connectionParameters;
+            this._cache = cache;
+            this._client = client;
+            this._connectionParameters = connectionParameters;
+            this._cacheSchema = cacheSchema;
+
+            this._storage = connectionParameters._options._storage;
+            this._refresherStateCache  = new RefresherStateCache(this._storage, `${rootKey}.RefresherStateCache`, 1000*3600);
+
+            this._lastTime = undefined;
+            this._buildNumber = undefined;
+            this._intervalId = null;
+        }
+
+        /**
+         * Start auto refresh
+         * @param {integer} refreshFrequency frequency of the refresh in ms ( default velue is 10 000 ms)
+         */
+        startAutoRefresh(refreshFrequency) {
+            if (!this._client.isLogged())
+              throw CampaignException.NOT_LOGGED_IN(undefined, `Cannot call startAutoRefresh: session not connected`);
+
+            if (this._intervalId != null) {
+                clearInterval(this._intervalId);
+            }
+            this._intervalId = setInterval(() => this._callAndRefresh(), refreshFrequency || 10000); // every 10 seconds by default
+        }
+
+        // Get last modified entities for the Campaign server and remove from cache last modified entities
+        _callAndRefresh() {
+            const that = this;
+            const soapCall = this._client._prepareSoapCall("xtk:session", "GetModifiedEntities", true, this._connectionParameters._options.extraHttpHeaders);
+
+            if (this._lastTime === undefined) {
+                const storedTime = this._refresherStateCache.get("time");
+                if (storedTime != undefined) {
+                    this._lastTime = storedTime;
+                }
+            }
+            if (this._buildNumber === undefined) {
+                const storedBuildNumber = this._refresherStateCache.get("buildNumber");
+                if (storedBuildNumber != undefined) {
+                    this._buildNumber = storedBuildNumber;
+                }
+            }
+
+            // Use Json because xtk:schema does not work directly in DomUtil.parse(`<cache buildNumber="9469" time="2022-06-30T00:00:00.000"><xtk:schema></xtk:schema></cache>`);
+            // due to the colon character
+            let jsonCache;
+            if (this._lastTime === undefined || this._buildNumber === undefined) {
+                jsonCache = {
+                    [this._cacheSchema]: {}
+                };
+            } else {
+                jsonCache = {
+                    buildNumber: this._buildNumber,
+                    lastModified: this._lastTime,
+                    [this._cacheSchema]: {}
+                };
+            }
+
+            const xmlDoc = DomUtil.fromJSON("cache", jsonCache, 'SimpleJson');
+            soapCall.writeDocument("script", xmlDoc);
```

Comment:
Why "script"?
The paramerer is actually called "cacheEntities" in https://git.corp.adobe.com/Campaign/ac/pull/13968/files#diff-cd14dce372b0e5aaf9d2734d3bb41fb2ed5c69c3d12858f539318bddbc5b96d0R430

Reply:
I think here the first parameter has no functional impact but as somewhere else, I think it's better to use "document"
`soapCall.writeDocument("document", xmlDoc);`

```
    /**
     * Sets a XML document-type parameter
     * @param {string} tag the parameter name
     * @param {Document} value the parameter value (XML document)
     */
    writeDocument(tag, document) {
        const node = this._addNode(tag, "", null, SOAP_ENCODING_XML);
        if (document !== null && document !== undefined) {
            const element = document.nodeType === 1 ? document : document.documentElement;
            const child = this._doc.importNode(element, true);
            node.appendChild(child);
        }
    }
```